riscv/sim.cc

```diff
@@ -74,9 +70,23 @@ sim_t::sim_t(const cfg_t *cfg, bool halted,
 
   debug_module.add_device(&bus);
 
+  socketif = NULL;
+#ifdef HAVE_BOOST_ASIO
+  if (socket_enabled) {
+    socketif = new socketif_t();
+  }
+#else
+  if (socket_enabled) {
+    fputs("Socket support requires compilation with boost asio; "
+         "please rebuild the riscv-isa-sim project without --without-boost-asio\n",
+         stderr);
+    abort();
+  }
+#endif
+
 #ifndef RISCV_ENABLE_DUAL_ENDIAN
   if (cfg->endianness != memif_endianness_little) {
-    fputs("Big-endian support has not been prroperly enabled; "
+    fputs("Big-endian support has not been properly enabled; "
```

Comment:
This sort of change should be in its own commit; refactoring commits should not have functional changes (even trivial ones).